editoast/src/modelsv2/projects.rs

```diff
@@ -0,0 +1,242 @@
+use crate::error::Result;
+use crate::modelsv2::Save;
+use crate::views::pagination::{Paginate, PaginatedResponse};
+use crate::DbPool;
+use actix_web::web::Data;
+use async_trait::async_trait;
+use chrono::{NaiveDateTime, Utc};
+use derivative::Derivative;
+use diesel::sql_types::BigInt;
+use diesel::{sql_query, ExpressionMethods, QueryDsl};
+use diesel_async::{AsyncPgConnection as PgConnection, RunQueryDsl};
+use editoast_derive::ModelV2;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+use crate::models::List;
+
+crate::schemas! {
+    Ordering,
+    ProjectWithStudies,
+    Project,
+    ProjectRow,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Derivative, ModelV2, ToSchema)]
+#[derivative(Default)]
+#[model(table = crate::tables::project)]
+#[model(row(derive(Debug, Clone, Serialize, ToSchema)))]
+pub struct Project {
+    pub id: i64,
+    pub name: String,
+    pub objectives: String,
+    pub description: String,
+    pub funders: String,
+    pub budget: i32,
+
+    #[schema(value_type = NaiveDateTime)]
+    pub creation_date: NaiveDateTime,
+
+    #[derivative(Default(value = "Utc::now().naive_utc()"))]
+    pub last_modification: NaiveDateTime,
+
+    #[derivative(Default(value = "Vec::new()"))]
+    #[schema(value_type = Vec<String>)]
+    pub tags: Vec<Option<String>>,
+
+    #[model(column = "image_id")]
+    #[schema(value_type = Option<i64>)]
+    pub image: Option<i64>,
+}
+
+#[derive(Debug, Clone, Serialize, QueryableByName, ToSchema)]
+pub struct ProjectWithStudies {
+    #[serde(flatten)]
+    #[diesel(embed)]
+    pub project: ProjectRow,
```

Comment:
we need also `QueryableByName` for `Project`

Reply:
My take on this is that PorjectWithStudies is not a real model but an endpoint response and that it should be move to *views/*. We can add a `Project::studies_count()` that we'd use to build `views::ProjectWithStudies`. Admittedly it's two queries, but:
1. We use the same connection object, so 🤷 
2. responsibility boundaries 

Otherwise, since models are not designed to be as precise as `QueryableByName` is, I think the "clean" way (or at least the most diesel way) to to this is to duplicate the fields. But again, to me, ProjectWithStudies is a view response.